plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -532,6 +569,20 @@ public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
     @Override
     public LocatedTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
     {
+        throw new UnsupportedOperationException("This method is not supported because getTableHandle with versions is implemented instead");
+    }
+
+    @Override
+    public LocatedTableHandle getTableHandle(
+            ConnectorSession session,
+            SchemaTableName tableName,
+            Optional<ConnectorTableVersion> startVersion,
+            Optional<ConnectorTableVersion> endVersion)
+    {
+        if (startVersion.isPresent()) {
+            throw new TrinoException(NOT_SUPPORTED, "Read table with start version is not supported");
```

Comment:
It would be nice to have test for that.

Reply:
I don't think we can add meaningful tests. The connectors always get empty start version. 
https://github.com/trinodb/trino/blob/5b2c7c729247f3db81c95627345e950334914afa/core/trino-parser/src/main/java/io/trino/sql/tree/QueryPeriod.java#L37-L48